pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/FilterParser.java

```diff
@@ -165,17 +168,17 @@ public FilterParser(FilterBuilder eval) {
      * @return the parsed filter
      * @throws Exception if the filter string had wrong syntax
      */
-    public Object parse(String filter) throws Exception {
+    public Object parse(byte[] filter) throws Exception {
         index = 0;
-        filterString = filter;
+        filterByteArr = filter;
         int opNumber;
 
         if (filter == null) {
             throw new FilterStringSyntaxException("filter parsing ended with no result");
         }
 
-        while (index < filterString.length()) {
-            char op = filterString.charAt(index);
+        while (index < filterByteArr.length) {
+            char op = (char) filterByteArr[index];
             ++index; // skip op character
             switch (op) {
                 case 'a':
```

Comment:
Should we have all conventional characters used in protocol as constants or enum?
'a','c', 's','d','o','l'